tests/storage/blocksd_test.py

```diff
@@ -1643,6 +1643,63 @@ def test_create_illegal_volume(domain_factory, domain_version, fake_task,
     assert vol.getLegality() == sc.ILLEGAL_VOL
 
 
+@requires_root
+def test_reduce_volume_called(domain_factory, fake_task, fake_sanlock):
+    sd_uuid = str(uuid.uuid4())
+    dom = domain_factory.create_domain(sd_uuid=sd_uuid, version=5)
+
+    img_uuid = str(uuid.uuid4())
+    vol_id = str(uuid.uuid4())
+    vol_capacity = 10 * GiB
+
+    dom.createVolume(
+        imgUUID=img_uuid,
+        capacity=vol_capacity,
+        volFormat=sc.COW_FORMAT,
+        preallocate=sc.SPARSE_VOL,
+        diskType=sc.DATA_DISKTYPE,
+        volUUID=vol_id,
+        desc="Base volume",
+        srcImgUUID=sc.BLANK_UUID,
+        srcVolUUID=sc.BLANK_UUID)
+
+    # Prepare volume before reducing
+    dom.produceVolume(img_uuid, vol_id).prepare()
+    dom.reduceVolume(img_uuid, vol_id)
+
+    # Check that (prepared) volume size has been reduced
+    assert dom.getVolumeSize(img_uuid, vol_id).truesize < vol_capacity
```

Comment:
This does not test that we reduced the volume to the optimal size.

To fix this test:
1. Create the volume with initial_size=5 * GiB
2. Check that the volume size after the reduce is vol.optimal_size()